File: sorting/bubblesort.py
def bubble_sort(array):
    has_swapped = True
    while (has_swapped):
        has_swapped = False
        i = 0
        while i < len(array) - 1:
            if array[i]> array[i + 1]:
                temp = array[i]
                array[i] = array[i+1]
                array[i+1] = temp
                has_swapped = True
            i += 1
array = [1,3,5,2,4]
bubble_sort((array))
print(array)

File: DSA/algorithms/bubbleSort.js
const swap = (arr, x, y, temp = arr[x]) => {
  arr[x] = arr[y];
  arr[y] = temp;
};
const arr = [1, 5, 2, 6, 3, 4, 7, 9, 8, 11];

let hasSwapped = true;
const bubbleSort = (arr) => {
  while (hasSwapped) {
    hasSwapped = false;
    arr.forEach((n, i) => {
      if (n > arr[i + 1]) {
        swap(arr, i, i + 1);
        hasSwapped = true;
      }
    });
  }
  return arr;
};

console.log(bubbleSort(arr));
